jobqueue/scheduler/kubernetes.go

```diff
@@ -308,58 +300,58 @@ func (s *k8s) reqCheck(req *Requirements) error {
 		CbChan: make(chan kubescheduler.Response),
 	}
 
-	s.Debug("Sending request to listener", "request", r)
+	clog.Debug(ctx, "Sending request to listener", "request", r)
 	go func() {
 		s.reqChan <- r
 	}()
 
-	s.Debug("Waiting on reqCheck to return")
+	clog.Debug(ctx, "Waiting on reqCheck to return")
 	resp := <-r.CbChan
 
 	s.esmutex.Lock()
 	defer s.esmutex.Unlock()
 
 	if resp.Error != nil {
-		s.Error("Requirements check received error", "error", resp.Error)
+		clog.Error(ctx, "Requirements check received error", "error", resp.Error)
 		s.es = resp.Ephemeral
 		return resp.Error
 	}
 
-	s.Debug("reqCheck returned ok")
+	clog.Debug(ctx, "reqCheck returned ok")
 	s.es = resp.Ephemeral
 
 	return resp.Error
 }
 
 // setMessageCallBack sets the given callback function.
-func (s *k8s) setMessageCallBack(cb MessageCallBack) {
-	s.Debug("setMessageCallBack called")
+func (s *k8s) setMessageCallBack(ctx context.Context, cb MessageCallBack) {
+	clog.Debug(ctx, "setMessageCallBack called")
```

Comment:
These debug lines are unnecessary, and therefore so is changing the callback methods to take a ctx?